cmake/modules/extensions.cmake

```diff
@@ -1335,7 +1335,10 @@ function(zephyr_linker_sources location)
     endif()
 
     # Find the relative path to the linker file from the include folder.
-    file(RELATIVE_PATH relpath ${ZEPHYR_BASE}/include ${path})
+    get_filename_component(ZEPHYR_BASE_REALPATH "${ZEPHYR_BASE}/include" REALPATH)
+    get_filename_component(PATH_REALPATH "${path}" REALPATH)
```

Comment:
The command `get_filename_component(... REALPATH)` has been superseded by `file(REAL_PATH ...)` cmake.
> `get_filename_component`
> 
> Get a specific component of a full filename.
>
> Changed in version 3.20: This command has been superseded by the [cmake_path()](https://cmake.org/cmake/help/latest/command/cmake_path.html#command:cmake_path) command, except for REALPATH, which is now offered by [file(REAL_PATH)](https://cmake.org/cmake/help/latest/command/file.html#real-path),

https://cmake.org/cmake/help/latest/command/get_filename_component.html#command:cmake_path

but please note that `file(REAL_PATH ...)` is affected by [CMP0152](https://cmake.org/cmake/help/latest/policy/CMP0152.html)., and before you ask, then yes, `get_filename_component(... REAL_PATH)` suffers the same issue related to symlink resolving, but doesn't have a policy for controlling this (always fixed at OLD behavior).

Which again shows that the use of symlinks can be tricky.

Also, variables intended for local use should be written in lowercase, like `path_realpath` and `zephyr_base_realpath`.
And please use another name than `zephyr_base_realpath`, as it is not the real path of Zephyr you are defining, but a subfolder (`include`).